### ISSUE TITLE
nanox build error: no method named `compare_exchange` found for struct `core::sync::atomic::AtomicU8` in the current scope

### ISSUE THREAD

**Issue by Pana** [LedgerHQ/ledger-device-rust-sdk#218]
When developing a Ledger device app with Rust, I introduced a fundamental library called bytes. This library is very foundational in Rust development and is widely used, including in embedded devices. However, when selecting the Nano X device, the compilation failed, and I encountered the following error.

However, when selecting the Nano S Plus, Stax, or Flex devices, the compilation succeeds.

![image](https://github.com/user-attachments/assets/4ff330ea-202c-48d5-be60-98eab9d6dd4c)

**Comment by Pana** [comment]
nanosplus, stax, flex build is ok

**Comment by Pana** [comment]
Related issue https://github.com/tokio-rs/bytes/issues/461

**Comment by yogh333** [comment]
Yes atomic swap is not supported on NanoX: https://github.com/LedgerHQ/ledger-device-rust-sdk/blob/63c08b167390ea38ed4a8709211481dd93b0eeb1/ledger_device_sdk/nanox.json#L4

**Comment by tdejoigny-ledger** [comment]
@Pana 
you can try to use `portable-atomic` crate by adding in `Cargo.toml` :
portable-atomic = { version = "1.10.0", features = ["unsafe-assume-single-core"] }

**Comment by ryankurte** [comment]
fyi this also requires a minor patch to `bytes`

https://github.com/tokio-rs/bytes/compare/master...ryankurte:rust-bytes:v1.4.0-portable-atomic

**Comment by Pana** [comment]
> Yes atomic swap is not supported on NanoX:
> 
> [ledger-device-rust-sdk/ledger_device_sdk/nanox.json](https://github.com/LedgerHQ/ledger-device-rust-sdk/blob/63c08b167390ea38ed4a8709211481dd93b0eeb1/ledger_device_sdk/nanox.json#L4)
> 
> Line 4 in [63c08b1](/LedgerHQ/ledger-device-rust-sdk/commit/63c08b167390ea38ed4a8709211481dd93b0eeb1)
> 
>  "atomic-cas": false,

Can we open this config atomic-cas ?

**Comment by Pana** [comment]
Thanks very much
